storage/maria/ma_unique.c

```diff
@@ -259,6 +259,8 @@ my_bool _ma_unique_comp(MARIA_UNIQUEDEF *def, const uchar *a, const uchar *b,
     {
       if (a_length != b_length)
         return 1;
+      if (pos_a == NULL || pos_b == NULL)
```

Comment:
I think UBSan should not fail if there is assertion like that:

```c++
DBUG_ASSERT(pos_a != NULL);
DBUG_ASSERT(pos_b != NULL);
```

Because it is certainly assertion, not the application logic. Moreover this is performance-critical code. I guess UBSan is usable only on debug build. I'd rather not change release code in favor of its warnings.

Reply:
`DBUG_ASSERT` will not have any impact on the program execution when `CMAKE_BUILD_TYPE=RelWithDebInfo` is being used. We should care what users and customers are running. For example with the current configuration of https://buildbot.mariadb.org/#/builders/588/ such assertions should have no effect.

However, it seems to me that we might want to declare the function parameters  `__attribute__((nonnull))`. As far as I can tell, `pos_a` or `pos_b` can only be null pointers if `a` or `b´ are:
```c
    pos_a= a+keyseg->start;
    pos_b= b+keyseg->start;
```
These assignments should already have triggered a runtime error for applying an offset to a null pointer.